### PR TITLE
#1386 fix faulty strings

### LINKED ISSUE #1386: Faulty snake cases in strings.xml
**Describe the bug**
There are tons of faulty snake cases in the `strings.xml` that need to be fixed. For example:
 `<string name="error_nozimfilesfound">` 

**Expected behavior**
Should be `error_no_zim_files_found` . Similarly, all faulty snake cases should be fixed in `strings.xml`.

### PR DESCRIPTION
**Pull request by jhbiggs**
Fixes #1386 

I believe I fixed it this time.  It was so simple after seeing @macgills 's clear direction.  I'm not sure what I was doing before.  Thanks for your patience.

### CODE REVIEW

**Comment by macgills** [comment]
Glad to be of assistance @jhbiggs ,
There is a compilation error right now
`continuous-integration/travis-ci/pr — The Travis CI build failed ` click on the `details` button and scroll to the bottom of the build (there is a circle in the top right of the build output window that does this for you) and scan up from the bottom to see what failed.
In this case it was 
```

> Task :app:compileKiwixDebugJavaWithJavac
2064
/home/travis/build/[secure]/[secure]-android/app/src/main/java/org/[secure]/[secure]mobile/main/MainActivity.java:2109: error: cannot find symbol
2065
          Snackbar.make(snackbarRoot, R.string.new_tab_snack_bar, Snackbar.LENGTH_LONG)
2066
                                              ^
2067
  symbol:   variable new_tab_snack_bar
2068
  location: class string
2069
Note: Some input files use or override a deprecated API.
2070
Note: Recompile with -Xlint:deprecation for details.
2071
Note: Some input files use unchecked or unsafe operations.
2072
Note: Recompile with -Xlint:unchecked for details.
2073
1 error
```

**Comment by macgills** [comment]
Many things have gone wrong here. The pre-commit hook did not run so lint did not run on your commit, this means that a) you did not build the project once before committing so it did not install b) you deleted the hook out of `.git/hooks`

Secondly you renamed `new_tab_snackbar` to `new_tab_snack_bar` in every locale but English. I am not sure how that is possible if you performed a rename action (shift+f6) on the resource? Regardless please change the resource name in English to the new form

**Comment by macgills** [comment]
Also a lot of branches were created for this pr
image
I thank you for following the branch naming format but these branches should also be under the folder `feature/jhbiggs`

**Comment by jhbiggs** [comment]
I am using Android Studio, and built the project there.  At the command line I added back the pre-commit file which I had deleted per suggestion in a previous chat.  Here's the output when I attempted to commit: 

Running lint...
.git/hooks/pre-commit: line 5: ./gradlew: Permission denied
.git/hooks/pre-commit: line 13: ./gradlew: Permission denied
Static analysis found violations and attempted to autofix, please commit these autoformat changes

Incidentally, build failed in Studio because a non-density apk was not found, so I disabled, then re-enabled it for the build to work on my machine.  

I'm not sure how to proceed here.  The multiple branches reflect my multiple attempts to work around build issues.  It's strange because such problems didn't arise with previous pull requests, and this string issue was so simple to fix, yet impossible for me to commit successfully.

**Comment by jhbiggs** [comment]
Some success here, fixed the snackbar change problem, figured out that I needed to run "gradle wrapper".  There was a string "article_count" which didn't appear to be used to the lint check, but was in one of the Kotlin files.  I took the suggestion from the command line output and added "abortOnError = false" to avoid the catch-22 there.

File: app/.idea/workspace.xml [hunk 1]
```diff
@@ -0,0 +1,397 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (macgills) [inline app/.idea/workspace.xml]: This file should not be here, why is there an `.idea` folder underneath `app` directory?

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,5 @@
-#Sun Jun 16 15:57:45 CDT 2019
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.6-bin.zip
```
Comment (macgills) [inline gradle/wrapper/gradle-wrapper.properties]: please revert this change, we should not be upgrading the projects gradle version for this task

File: gradlew [hunk 3]
```diff
@@ -1,5 +1,21 @@
 #!/usr/bin/env sh
 
+#
```
Comment (macgills) [inline gradlew]: this file should not have been modified, what OS are you running?

File: gradlew.bat [hunk 4]
```diff
@@ -1,3 +1,19 @@
+@rem
+@rem Copyright 2015 the original author or authors.
```
Comment (macgills) [inline gradlew.bat]: there also should have been no updates to this

**Comment by macgills** [comment]
@jhbiggs there are now conflicts so please merge develop into your branch and resolve them.

You should not disable lint just so you are able to commit, lint will run on our CI server on travis and if it fails there then it is a waste of the 20/30mins it takes to get to that build phase. Lint is necessary to our project's quality, if it was misbehaving for you the appropriate thing to do is to suppress the error, more details [here](https://developer.android.com/studio/write/lint#configuring-lint-checking-in-xml).

If the permission was denied for your `gradlew` then you should have [changed the permission](https://stackoverflow.com/a/17669566), it is weird that this hasn't already happened as a part of installing the git hook is that it changes the permission on it when on linux/mac.

I'll create a hotfix for `build failed in Studio because a non-density apk was not found`

**Comment by macgills** [comment]
@jhbiggs it is fine, this is a valuable learning exercise. Android Development is a very dense topic and to get into it there are many many hurdles. Integrating with code quality tooling can be rough and I have definitely set this up as "windows first", though I don't think we have had as many pain points with Linux.

It might be best to open a new branch of develop and redo the work with the knowledge you have now about how pre-commit hooks and gradlew works. 

What are you using for git? Commandline? AndroidStudio might be more friendly if you use the version control tab. Personally I use Sourcetree because I like the ui best. 

The only files that should change for this are strings.xml and the java files that use them.

If changing the permission on gradlew doesn't work then you can skip the pre-commit hook, android studio gives you a handy checkbox to untick or you can add `--no-verify` to `git commit` and rely on the ci linter

**Comment by macgills** [comment]
@jhbiggs maybe reinstall Android Studio? You should be able to open the main folder and have all the appropriate options. I wonder if any other MacOs users

"remove redundant imports" is likely a result of many project settings now being included in git, auto optimising imports on the fly is absolutely an option that should be enabled on Android Studio.

Try using the version control tab in AS for at least 1 commit, it is of great importance that you only commit files you actually changed. If you are confused about why a file is in a commit you can communicate with me here or via slack.
